.test-infra/dataproc/flink_cluster.sh

```diff
@@ -137,12 +134,29 @@ function create_cluster() {
   gcloud dataproc clusters create $CLUSTER_NAME --num-workers=$num_dataproc_workers --initialization-actions $DOCKER_INIT,$BEAM_INIT,$FLINK_INIT --metadata "${metadata}", --image-version=$image_version --zone=$GCLOUD_ZONE --quiet
 }
 
-function main() {
+# Runs init actions for Docker, Portability framework (Beam) and Flink cluster
+# and opens an SSH tunnel to connect with Flink easily and run Beam jobs.
+function create() {
   upload_init_actions
   create_cluster
   get_leader
   [[ -n "${JOB_SERVER_IMAGE:=}" ]] && start_job_server
   start_tunnel
 }
 
-main "$@"
\ No newline at end of file
+# Recreates a Flink cluster.
+function restart() {
+  delete
+  create
+}
+
+# Deletes a Flink cluster.
+function delete() {
+  gcloud dataproc clusters delete $CLUSTER_NAME --quiet
+}
+
+function scale() {
```

Comment:
Why don't you use `gcloud dataproc update` for scaling the dataproc cluster? 

https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/scaling-clusters

Reply:
Did that cause exception mentioned in the commit name? Rebooting the cluster completely seems radical - is that the only option?

Reply:
Further discussion is here: https://github.com/apache/beam/pull/9381